Sources/CryptomatorCloudAccess/Box/BoxAuthenticator.swift

```diff
@@ -20,31 +20,28 @@ public enum BoxAuthenticatorError: Error {
 }
 
 public enum BoxAuthenticator {
-	public static let sdk = BoxSDK(clientId: BoxSetup.constants.clientId, clientSecret: BoxSetup.constants.clientSecret)
+	public static func authenticate(from viewController: UIViewController, tokenStorage: TokenStorage) -> Promise<BoxCredential> {
+		let pendingPromise = Promise<BoxCredential>.pending()
 
-	public static func authenticate(from viewController: UIViewController, tokenStore: TokenStore) -> Promise<(BoxClient, String)> {
-		return Promise { fulfill, reject in
+		_Concurrency.Task {
+			do {
+				guard let context = viewController as? ASWebAuthenticationPresentationContextProviding else {
+					throw BoxAuthenticatorError.invalidContext
+				}
 
-			guard let context = viewController as? ASWebAuthenticationPresentationContextProviding else {
-				reject(BoxAuthenticatorError.invalidContext)
-				return
-			}
+				let config = OAuthConfig(clientId: BoxSetup.constants.clientId, clientSecret: BoxSetup.constants.clientSecret, tokenStorage: tokenStorage)
+				let oauth = BoxOAuth(config: config)
 
-			sdk.getOAuth2Client(tokenStore: tokenStore, context: context) { result in
-				switch result {
-				case let .success(client):
-					client.users.getCurrent(fields: ["id"]) { userResult in
-						switch userResult {
-						case let .success(user):
-							fulfill((client, user.id))
-						case .failure:
-							reject(BoxAuthenticatorError.authenticationFailed)
-						}
-					}
-				case .failure:
-					reject(BoxAuthenticatorError.authenticationFailed)
-				}
+				// Run the login flow and store the access token using tokenStorage
+				try await oauth.runLoginFlow(options: .init(), context: context)
+				// TODO: Catch error when login failed
```

Comment:
Just wanted to clarify here that I meant "when login is being cancelled". I just assumed that `runLoginFlow` would throw a special error in that case.

Reply:
Implemented in commit d544b20.